### PR TITLE
Auto deployment for windows using appveyor

### PR DESCRIPTION
**Pull request by syxolk**
- Installs `pandoc` and the python `toml` library.
- Set `RUSTFLAGS` early to make it available in `test_script` and `build_script`
- Use PowerShell as much as possible:
  - To insert the current version in `indentex_template.wxs`
  - ~To rename the installer from `indentex.msi` to `indentex_[version]_[arch].msi`~
- Use Python to get the package version from `Cargo.toml`

### CODE REVIEW

File: packaging/windows_wix/package_info.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from __future__ import print_function
+from os import path
+import toml
+
+
+def get_version():
+    with open(path.join("..", "..", "Cargo.toml")) as f:
+        cargo = toml.load(f)
+        return cargo["package"]["version"]
```
Comment (mp4096) [inline packaging/windows_wix/package_info.py]: I think this should be like this:

```python
def get_version():
    with open(path.join("..", "..", "Cargo.toml")) as f:
        cargo = toml.load(f)
    return cargo["package"]["version"]
```

Then the context is closed explicitly before returning the value.

**Comment by mp4096** [comment]
First of all, thanks for the PR and all the work! This was a really nasty one. :nauseated_face:

Some questions:
* Is it necessary to have a `.gitignore` in the packaging folder? Personally, I prefer a single `.gitignore` for the whole repo.
* Is it possible to avoid the Python / TOML dependency and parse (with regexen?) the output of `indentex -V`?
* I think it is more prudent to perform `--release` builds and install `pandoc` and `toml` only on tag events, in order to load AppVeyor as little as possible. Check out this branch: https://github.com/mp4096/indentex/tree/check-if-tag

**Comment by mp4096** [comment]
> Is it possible to avoid the Python / TOML dependency and parse (with regexen?) the output of `indentex -V`?

E.g. we could replace all whitespaces with an underscore and append `_amd64` at the end.

**Comment by syxolk** [comment]
> Is it possible to avoid the Python / TOML dependency and parse (with regexes?) the output of indentex -V?

Yes, Powershell has an integrated regex engine!

> E.g. we could replace all whitespaces with an underscore and append _amd64 at the end.

We still need the clean version number to replace it in `indentex_template.wxs`.

> Is it necessary to have a .gitignore in the packaging folder? Personally, I prefer a single .gitignore for the whole repo.

It felt more natural to put the `.gitignore` there but I could move it.

**Comment by mp4096** [comment]
>> E.g. we could replace all whitespaces with an underscore and append `_amd64` at the end.
>
> We still need the clean version number to replace it in `indentex_template.wxs`.

Oh yeah, I didn't think about this.

---

I'd prefer a single `.gitignore` file. :upside_down_face:

**Comment by mp4096** [comment]
Great! :+1: 

Thank you so much, this PR surely caused a lot of frustration.

**Pull request merged**